File: 30days/Day1_Arrays/repeatAndMissing.cpp
// Find Repeat and Missing 
#include<bits/stdc++.h>
using namespace std;
typedef long long int ll;

class Solution{
	public:
		int *findTwoElement(int *arr,int n){
			static int ans[2]={-1,-1};
			int sqr=0,sm=0,actSm=((n*(n+1))/2),actsqr=((n*(n+1)*(2*n+1))/6);
			for(int i=0;i<n;i++){
				sm+=arr[i];
				sqr+=arr[i]*arr[i];
			}
			int dif,sqdif;
			if (actSm>sm){
				dif=actSm-sm;
				sqdif=(actsqr-sqr);
				ans[1]=(dif+(sqdif/dif))/2;
				ans[0]=ans[1]-dif;
			}
			else{
				dif=sm-actSm;
				sqdif=sqr-actsqr;
				ans[0]=(dif+(sqdif/dif))/2;
				ans[1]=ans[0]-dif;
			}
			return ans;
		}
};
int main(){
	int n;
	cin>>n;
	int arr[n];
	for (int i=0;i<n;i++){
		cin>>arr[i];
	}
	Solution S;
	int *ans=S.findTwoElement(arr,n);
	cout<<ans[0]<<" "<<ans[1]<<endl;
}

File: 30days/Day1_Arrays/repeatAndMissing.py
#Find Repeat and Missing Number
class Solution():
    def findTwoElement(self,arr,n):
        xr1,xr2,xr3=0,0,0
        sm,actsm=0,(n*(n+1))//2
        for i in range(n):
            sm+=arr[i]
            xr1=xr1^(i+1)^arr[i];
        small=1
        while xr1&1==0:
            small*=2 
            xr1=xr1>>1
        for i in range(n):
            if small&(i+1):
                xr2=xr2^(i+1) 
            else:
                xr3=xr3^(i+1) 
            if small&arr[i]:
                xr2=xr2^arr[i]
            else:
                xr3=xr3^arr[i] 
        if sm>actsm:
            if xr3>xr2:
                return (xr3,xr2) 
            else:
                return (xr2,xr3) 
        else:
            if xr3>xr2:
                return (xr2,xr3)
            else:
                return (xr3,xr2) 

